packages/fx-core/src/component/generator/templates/templateGenerator.ts

```diff
@@ -0,0 +1,223 @@
+    const variables = this.getDefaultReplaceMap(inputs);
+    return Promise.resolve(ok([{ templateName, language, variables }]));
+  }
+
+  @hooks([
+    ActionExecutionMW({
+      enableProgressBar: true,
+      progressTitle: ProgressTitles.create,
+      progressSteps: 1,
+      enableTelemetry: true,
+      telemetryEventName: TelemetryEvent.GenerateTemplate,
+    }),
+  ])
+  public async run(
+    ctx: Context,
+    inputs: Inputs,
+    destinationPath: string,
+    actionContext?: ActionContext
+  ): Promise<Result<undefined, FxError>> {
+    const preResult = await this.getTemplateInfos(ctx, inputs, actionContext);
+    if (preResult.isErr()) return err(preResult.error);
+
+    const templateInfos = preResult.value;
+    for (const templateInfo of templateInfos) {
+      await this.scaffolding(ctx, templateInfo, destinationPath, actionContext);
+    }
+
+    const postRes = await this.post(ctx, inputs, destinationPath, actionContext);
+    if (postRes.isErr()) return postRes;
+
+    return ok(undefined);
+  }
+
+  public post(
+    ctx: Context,
+    inputs: Inputs,
+    destinationPath: string,
+    actionContext?: ActionContext
+  ): Promise<Result<undefined, FxError>> {
+    return Promise.resolve(ok(undefined));
+  }
+
+  public getDefaultReplaceMap(inputs: Inputs): { [key: string]: string } {
+    const appName = inputs[QuestionNames.AppName] as string;
+    const safeProjectName =
+      inputs[QuestionNames.SafeProjectName] ?? convertToAlphanumericOnly(appName);
+    const targetFramework = inputs.targetFramework;
+    const placeProjectFileInSolutionDir = inputs.placeProjectFileInSolutionDir === "true";
+    const llmService: string | undefined = inputs[QuestionNames.LLMService];
+    const openAIKey: string | undefined = inputs[QuestionNames.OpenAIKey];
+    const azureOpenAIKey: string | undefined = inputs[QuestionNames.AzureOpenAIKey];
+    const azureOpenAIEndpoint: string | undefined = inputs[QuestionNames.AzureOpenAIEndpoint];
+
+    return {
+      appName: appName,
+      ProjectName: appName,
+      TargetFramework: targetFramework ?? "net8.0",
+      PlaceProjectFileInSolutionDir: placeProjectFileInSolutionDir ? "true" : "",
+      SafeProjectName: safeProjectName,
+      SafeProjectNameLowerCase: safeProjectName.toLocaleLowerCase(),
+      enableTestToolByDefault: enableTestToolByDefault() ? "true" : "",
+      useOpenAI: llmService === "llm-service-openai" ? "true" : "",
+      useAzureOpenAI: llmService === "llm-service-azure-openai" ? "true" : "",
+      openAIKey: openAIKey ?? "",
+      azureOpenAIKey: azureOpenAIKey ?? "",
+      azureOpenAIEndpoint: azureOpenAIEndpoint ?? "",
+      isNewProjectTypeEnabled: isNewProjectTypeEnabled() ? "true" : "",
+      NewProjectTypeName: process.env.TEAMSFX_NEW_PROJECT_TYPE_NAME ?? "TeamsApp",
+      NewProjectTypeExt: process.env.TEAMSFX_NEW_PROJECT_TYPE_EXTENSION ?? "ttkproj",
+    };
+  }
+
+  private getTemplateName(inputs: Inputs) {
```

Comment:
How about moving this function into `templateNames.ts`, so that when onboarding new templates this `templateGenerator.ts` can remain unchanged.

Reply:
Good idea. I agree that we should decouple the `getTemplateName` logic from generator, and I plan to refactor this logic in next PR.

Reply:
+1